### ISSUE TITLE
Call stack exceeded when creating large collections

### ISSUE THREAD

**Issue by joebalancio** [mio/mio#19]
## Scenario 1
```js
var data = []; // 2000 items
Resource.Collection.post(data);
```
Error:
```
lib/resource.js:933
   var args = Array.prototype.slice.call(arguments, 1);
                                    ^
 RangeError: Maximum call stack size exceeded
```

## Scenario 2
```
var data = []; // 2000 items
var collection = new Resource.Collection(data);
collection.post();
```
Error:
```
 lib/resource.js:0
 (function (exports, require, module, __filename, __dirname) { var Collection =
 ^
 RangeError: Maximum call stack size exceeded
```

**Comment by joebalancio** [comment]
## Scenario 3
When sending over HTTP the response is
```js
{
    "name": "RangeError",
    "type": "stack_overflow",
    "message": "Maximum call stack size exceeded"
}
```

**Comment by alexmingoia** [comment]
Fixed in `mio-validators@1.1.3`
